### PR TITLE
Authz is topic protected

### PR DESCRIPTION
**Pull request by craig8**

### CODE REVIEW

File: src/volttron/types/auth/auth_service.py [hunk 1]
```diff
@@ -230,6 +244,21 @@ def create_protected_topics(self, *, topic_name_patterns: list[str]) -> bool:
     def remove_protected_topics(self, *, topic_name_patterns: list[str]) -> bool:
         ...
 
+    @abstractmethod
+    def is_protected_topic(self, *, topic_expression: str) -> bool:
```
Comment (schandrika) [inline src/volttron/types/auth/auth_service.py:248]: duplicate declaration

File: tests/unit/input.json [hunk 2]
```diff
@@ -1,101 +0,0 @@
-{
```
Comment (schandrika) [inline tests/unit/input.json:1]: why is this file deleted?

File: tests/unit/test_authz_model.py [hunk 3]
```diff
@@ -1,807 +0,0 @@
-from pathlib import Path
```
Comment (schandrika) [inline tests/unit/test_authz_model.py:1]: why is this file deleted?

File: src/volttron/types/auth/auth_service.py (hunk 1 above)
Comment (schandrika) [reply src/volttron/types/auth/auth_service.py:248]: Got confused by the difference in parameter name.

File: tests/unit/test_authz_model.py [hunk 4]
```diff
@@ -805,3 +732,30 @@ def test_remove_agents_from_group():
...
+    # else topic_prefix.*
+    non_regex_topic = "devices/building1"
+    assert authz_map.create_protected_topics(topic_name_patterns=[non_regex_topic])
+
+    # prefix match
+    assert authz_map.is_protected_topic(topic_name_pattern=non_regex_topic)
+    assert authz_map.is_protected_topic(topic_name_pattern=non_regex_topic + "false")
+    assert not authz_map.is_protected_topic(topic_name_pattern=non_regex_topic[:-1])
+
+    reg_ex_topic = "/devices/building2/*/"
```
Comment (schandrika) [inline tests/unit/test_authz_model.py:756]: shouldn't the topic pattern be ".*" to be valid?

File: src/volttron/server/aip.py [hunk 5]
```diff
@@ -681,8 +667,17 @@ def install_agent_source(self, agent: str, force: bool = False, pre_release: boo
             # In such case, we should be able to revert to current version - so explicitly find current version and
             # see if that can be installed from pypi because local wheel might not be there anymore.
             try:
-                cmd_dry_run = ["poetry", "--directory", self._server_opts.poetry_project_path.as_posix(), "add",
-                               f"{agent_name}=={current_version}", "--dry-run"]
+                if agent.endswith(".whl"):
```
Comment (schandrika) [inline src/volttron/server/aip.py:670]: @craig8 why is there a dry run with new agent whl? Here all we are trying to do is see if we can restore the currently installed version of agent in case the install of the pass "agent" (.whl or otherwise) fails

Comment (craig8) [reply src/volttron/server/aip.py:670]: So this is part of the issue that I was talking about.  I will be dealing with this in a later PR.

File: tests/unit/test_authz_model.py (hunk 4 above)
Comment (craig8) [reply tests/unit/test_authz_model.py:756]: This is valid it works out just as it is and follows the correct chain of matches.
